src/rust/engine/process_execution/src/lib.rs

```diff
@@ -153,3 +158,39 @@ impl CommandRunner for BoundedCommandRunner {
     self.inner.1.with_acquired(move || inner.0.run(req))
   }
 }
+
+#[cfg(test)]
+mod tests {
+  use super::ExecuteProcessRequest;
+  use std::collections::hash_map::DefaultHasher;
+  use std::collections::{BTreeMap, BTreeSet};
+  use std::hash::Hash;
+
+  #[test]
+  fn execute_process_request_equality() {
+    let a = ExecuteProcessRequest {
+      argv: vec![],
+      env: BTreeMap::new(),
+      input_files: hashing::EMPTY_DIGEST,
+      output_files: BTreeSet::new(),
+      output_directories: BTreeSet::new(),
+      timeout: std::time::Duration::new(0, 0),
+      description: format!("One message"),
+      jdk_home: None,
+    };
+
+    let b = ExecuteProcessRequest {
+      argv: vec![],
+      env: BTreeMap::new(),
+      input_files: hashing::EMPTY_DIGEST,
+      output_files: BTreeSet::new(),
+      output_directories: BTreeSet::new(),
+      timeout: std::time::Duration::new(0, 0),
+      description: format!("Different message"),
+      jdk_home: None,
+    };
```

Comment:
In Rust, can you use a partial function to reduce the duplication? Like this:

```node
const createProcessRequest = (description: str) => ExecuteProcessRequest([], BTree(), ..., description)
```

Easiest way to do this is probably an inner function like this https://stackoverflow.com/a/26685687.

Then your test can be:

```rust
fn execute_process_request_equality() {
  fn create_execute_process_request(description: str) {
    ...
  }

 let a = create_execute_process_request("One message");
 let b =  create_execute_process_request("Different message");
 
 assertions;
}
```